oxenstored/store.ml

```diff
@@ -142,9 +142,7 @@ module Path = struct
     || c = '-'
     || c = '@'
 
-  let name_is_valid name =
-    name <> ""
-    && String.fold_left (fun accu c -> accu && char_is_valid c) true name
+  let name_is_valid name = name <> "" && String.for_all char_is_valid name
```

Comment:
If we care about performance `char_is_valid` could be implemented as a lookup table; possibly even using 256 bits.

Reply:
We would need to measure if that's worth it, but it's sad to see that OCaml can't turn the equivalent match code into something nicer than a bunch of jumps, compare [Rust](https://godbolt.org/z/7c7PxGszW) vs. [Flambda](https://godbolt.org/z/xGG1hcefh)

Reply:
You could use `char_is_valid` to build that bitmap lookup table, and then call `char_is_valid_lookup` instead. That way we can still tweak `char_is_valid` easily.

The compiler could also be improved, but that is another topic.